routers/web/repo/release.go

```diff
@@ -98,7 +98,12 @@ func releasesOrTags(ctx *context.Context, isTagList bool) {
 		listOptions.PageSize = setting.API.MaxResponseItems
 	}
 
-	tags, err := ctx.Repo.GitRepo.GetTags(listOptions.GetStartEnd())
+	tagListStart, tagListEnd := listOptions.GetStartEnd()
+	if !isTagList {
+		// tags are used for compare feature witch needs all tags
+		tagListStart, tagListEnd = 0, 0
+	}
```

Comment:
Reversing the logic, so you don't unnecessary call `GetStartEnd()`
```suggestion
	tagListStart, tagListEnd := 0, 0
	if isTagList {
		tagListStart, tagListEnd = listOptions.GetStartEnd()
	}
```

Reply:
this should not make a difference but I'll ajust ...
beside we should have it fixed later properly (#20073)

Reply:
> this should not make a difference but I'll ajust ...

Eh, seems it's better code-taste to use default values and then maybe the adjusted values. But feel free to ignore(really should prefix these comments with `code-style nits`)

Reply:
done